stock_warehouse_flow_delivery_refresh/models/stock_move.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2025 Camptocamp SA, BCIM
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _stock_warehouse_flow_onchange_carrier(self):
+        flow_model = self.env["stock.warehouse.flow"]
+        moves_need_release = self.filtered(lambda m: m.picking_id.need_release)
+        for move in moves_need_release:
+            flow_model._search_and_apply_for_move(move, assign_picking=True)
```

Comment:
By moving the check against `need_release` into the for loop, you would not need a second iteration. 
Also the `need_release` is stored on the move it self. 
```suggestion
        for move in self:
            if  not move.need_release:
                continue
            flow_model._search_and_apply_for_move(move, assign_picking=True)
```

Reply:
makes sense, thanks!